src/script/lua_api/l_vmanip.cpp

```diff
@@ -260,11 +260,17 @@ int LuaVoxelManip::l_get_light_data(lua_State *L)
 	NO_MAP_LOCK_REQUIRED;
 
 	LuaVoxelManip *o = checkobject(L, 1);
+	bool use_buffer  = lua_istable(L, 2);
+
 	MMVManip *vm = o->vm;
 
 	u32 volume = vm->m_area.getVolume();
 
-	lua_createtable(L, volume, 0);
+	if (use_buffer)
+		lua_pushvalue(L, 2);
```

Comment:
Is there an option to reserve places in the table, like done in `lua_createtable`? Resizing the table might cause a performance drop.

Reply:
Not that I know of. This is just how `get_data` and `get_param2_data` do it.

Reply:
I see. The only option would be to use `luaH_resizearray`, which is unfortunately not exposed anywhere.

Reply:
Also, after a buffer is used with one of these methods once, it will not have to be resized in subsequent calls.